indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/HttpRemoteTaskRunner.java

```diff
@@ -1108,20 +1071,18 @@ public void registerListener(TaskRunnerListener listener, Executor executor)
 
     final Pair<TaskRunnerListener, Executor> listenerPair = Pair.of(listener, executor);
 
-    synchronized (statusLock) {
-      for (Map.Entry<String, HttpRemoteTaskRunnerWorkItem> entry : tasks.entrySet()) {
-        if (entry.getValue().getState() == HttpRemoteTaskRunnerWorkItem.State.RUNNING) {
-          TaskRunnerUtils.notifyLocationChanged(
-              ImmutableList.of(listenerPair),
-              entry.getKey(),
-              entry.getValue().getLocation()
-          );
-        }
+    tasks.forEach((taskId, taskEntry) -> {
```

Comment:
Avoid using `.forEach()` since it is not mutually exclusive with `.compute()` even on the same key.